src/main/java/com/fasterxml/jackson/databind/deser/std/EnumDeserializer.java

```diff
@@ -425,4 +446,49 @@ protected boolean useDefaultValueForUnknownEnum(DeserializationContext ctxt) {
           && (Boolean.TRUE.equals(_useDefaultValueForUnknownEnum)
           || ctxt.isEnabled(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE));
     }
+
+    /**
+     * Checks wheather current Enum class is annotated with
+     * {@link com.fasterxml.jackson.databind.annotation.EnumNaming}
+     *
+     * @since 2.15
+     */
+    private boolean _hasEnumNaming(DeserializationContext ctxt) {
+        Boolean exists = _hasEnumNaming;
+        if (exists == null) {
+            synchronized (this) {
+                exists = _hasEnumNaming;
+                if (exists == null) {
+                    exists = _getEnumNamingLookup(ctxt) != null;
+                    _hasEnumNaming = exists;
+                }
+            }
+        }
+        return exists;
+    }
+
+    /**
+     * Returns the CompactStringObjectMap used for enum name lookup of naming strategy.
+     *
+     * @since 2.15
+     */
+    protected CompactStringObjectMap _getEnumNamingLookup(DeserializationContext ctxt) {
+        CompactStringObjectMap lookup = _lookupByEnumNaming;
+        if (lookup == null) {
+            synchronized (this) {
+                lookup = _lookupByEnumNaming;
+                if (lookup == null) {
+                    EnumNamingStrategy namingStrategy = EnumPropertiesCollector
```

Comment:
I think this should be fetched via `AnnotationIntrospector`, directly. Or, if we prefer, via `DeserializationContext`.